### DISCUSSION TITLE
Validate the schema itself

### DISCUSSION THREAD

**Discussion by elrandira** [python-jsonschema/jsonschema#1103] [General]
Hi,

in a project of my company, the Draf7Validator is used and I noticed some errors in the schema itself are not detected in our current use. We have a JSON schema definition where the `required` property was placed inside the `properties` property instead of outside.

https://json-schema.org/understanding-json-schema/reference/object.html#required

Is there a way to detect such errors?

Or I suppose it would require to run jsonschema on the schema itself but a _jsonschema of the jsonschema_ is needed here.

``` 
        "architecture": {
            "type": "object",
            "properties": {
                "variant": {
                    "description": "Customer specific setting.",
                    "type": "string",
                    "enum": [
                        "Default",
                        "TypeA",
                        "TypeB",
                        "TypeC"
                    ],
                    "default": "TypeB"
                },
                "required": [   <== misplaced property
                    "variant"
                ]
            }
        }
```

**Comment by klappj** [comment]
I'm looking at the same issue; trying to validate a schema itself.

**Comment by eoriont** [comment]
Maybe meta-schemas are what you want? https://json-schema.org/specification.html#meta-schemas

**Reply by elrandira** [reply]
Thanks for the tip, I'll try to have a look tomorrow

**Reply by elrandira** [reply]
I might be in the correct direction but that would not be enough. 

I think the properties/properties definition should evolve to exclude elements of properties. Something like this for the schema7:

```
{
    "$schema": "http://json-schema.org/draft-07/schema#",
    "$id": "http://json-schema.org/draft-07/schema#",
    "title": "Core schema meta-schema",
    "definitions": {
        < .. removed for clarity .. >
    },
    "type": ["object", "boolean"],
    "properties": {
        < .. removed for clarity .. >
        "properties": {
            "type": "object",
            "additionalProperties": { "$ref": "#" },
            "default": {},
            "not": {                <== this is added
                "anyOf": [
                    { "$ref": "#/properties" }
                ]
            }
        },
       < .. removed for clarity .. >
    },
    "default": true
}
```
